File: dawii_20222_miercoles_frontend_grupo_04-master/src/app/services/alumno.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { AppSettings } from '../app.settings';
import { Alumno } from '../models/alumno.model';

const baseUrlAlumno = AppSettings.API_ENDPOINT+ '/alumno';
const baseUrlCrudAlumno = AppSettings.API_ENDPOINT+ '/crudAlumno';


@Injectable({
  providedIn: 'root'
})
export class AlumnoService {

  constructor(private http : HttpClient) { }

  registroAlumno(data: Alumno): Observable<any>{
    return this.http.post(baseUrlAlumno, data)
  }

  registroAlumnoCrud(data: Alumno): Observable<any>{
    return this.http.post(baseUrlCrudAlumno, data)
  }


  actualizaAlumnoCrud(data: Alumno): Observable<any>{
    return this.http.put(baseUrlCrudAlumno + "/actualizaAlumno", data)
  }

  eliminaAlumnoCrud(idAlumno: number): Observable<any>{
    return this.http.delete(baseUrlCrudAlumno + "/eliminaAlumno/"+idAlumno)
  }

  consultaAlumno(nombre:string, apellido:string, dni:string, idPais: number, estado: number): Observable<any>{

    const params = new HttpParams().set("nombre", nombre).set("apellido", apellido).set("dni", dni).set("idPais", idPais).set("estado", estado);  

    return this.http.get(baseUrlAlumno + "/listaAlumnoConParametros", {params});

  }

  consultaAlumnoCrud(nombre:string, apellido:string, dni:string, idPais: number, estado: number): Observable<any>{

    const params = new HttpParams().set("nombre", nombre).set("apellido", apellido).set("dni", dni).set("idPais", idPais).set("estado", estado);  

    return this.http.get(baseUrlCrudAlumno + "/listaAlumnoConParametros", {params});

  }

  obtenerAlumnoCrud(idAlumno:number): Observable<any>{
    return this.http.get(baseUrlCrudAlumno+"/busca/"+idAlumno);
  }

}

File: dawii_20222_miercoles_backend_grupo_04-master/src/main/java/com/cibertec/service/AlumnoService.java
package com.cibertec.service;

import java.util.List;
import java.util.Optional;

import com.cibertec.entity.Alumno;

public interface AlumnoService {
	
	public abstract Alumno insertarAlumno(Alumno obj);
	
	public abstract List<Alumno> listaTodos();
	
	public abstract List<Alumno> listaAlumnoDni(String dni);

	public List<Alumno> listaAlumnoDniEstadoIdPais(String nombre, String apellido, String dni,  int estado, int idPais);
	
	public abstract Alumno actualizaAlumno(Alumno obj);
	
	public abstract void eliminaAlumno(int idAlumno);
	
	public abstract List<Alumno> listaAlumnoDniDiferenteDelMismoID(String dni, int id);

	public Optional<Alumno> buscarAlumnoPorId(int id);
	
	public Alumno buscarAlumnoPorIdAlumno(int idAlumno);

	
}
